### PR TITLE
Domain decomposition example using LDC

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
A simple example showing how to use domain decomposition using X-PINN style and FBPINN style in Modulus. The concept is demonstrated on a Lid-Driven Cavity case.

Domain is divided as shown below:

image

For FBPINN, the basis functions are used as below:

image

Results:

image

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus-sym/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus-sym/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus-sym/issues) is linked to this pull request.

### CODE REVIEW

**Comment by mnabian** [comment]
@ktangsali in my opinion this example should demonstrate how the training workload for each subdomain can be distributed on different GPUs, which does not exist currently in your example.
Also, if for any reason, you are interested in single GPU training with multiple domains, consider the FBPINNs approach, which essentially helps you get rid of the interface constraints. Your example can be easily extended to FBPINNs by removing the interface constraints, and switching from Heaviside to a smooth function (e.g. hyperbolic tangent) which overlaps with the opposite domain.

**Review by mnabian — CHANGES_REQUESTED** [review]

**Comment by mnabian** [comment]
Because the distributed training component is missing here, should we call this example something other than XPINNs? Maybe domain decomposition?

**Review by mnabian — APPROVED** [review]

File: examples/ldc/ldc_2d_xpinn.py [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    flow_net_2 = instantiate_arch(
+        input_keys=[Key("x"), Key("y")],
+        output_keys=[Key("u_2"), Key("v_2"), Key("p_2")],
+        cfg=cfg.arch.fully_connected,
+    )
+
+    from sympy import Piecewise, Heaviside, tanh
+    from modulus.sym.node import Node
+
+    # nodes for interface condition. Currently only dirichlet, but can be extended to neumann too
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply examples/ldc/ldc_2d_xpinn.py]: Thanks! Yes, definitely it is possible to add that, however for this example, just the solution continuity suffices so I have added only that. I have also added comments to reflect the same.

**Comment by ktangsali** [comment]
Thanks @mnabian , I have addressed your comments and also renamed the example. I think these two examples are a great demonstration of just how to use different neural networks in different domains of the problem and couple them via interface/basis functions. The distributed part is a nice-to-have and we can add that in a separate PR.

**Pull request merged**